### PR TITLE
use a sane version number for the client

### PR DESCRIPTION
**Pull request by jtnord**
Use a version number that maps to the k8s client for sane versioning.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -13,7 +13,7 @@
     <version>${revision}${changelist}</version>
     <packaging>hpi</packaging>
     <properties>
-        <revision>1.0.1</revision>
+        <revision>4.4.2-1</revision>
```
Comment (jtnord) [inline pom.xml]: was not sure if incrementals could handle `<revision>${kubernetes-client.version}-1</revision>` which would be cleaner
//cc @jglick

Comment (jglick) [reply pom.xml]: Unfortunately no: Maven hard-codes a list of properties which are allowed in this context.

**Comment by jglick** [comment]
I think the idea behind using a more generic version number was that at some point certain Jenkins-specific API classes would be moved into this plugin, so that it would not be a pure library wrapper.

**Comment by jtnord** [comment]
> I think the idea behind using a more generic version number was that at some point certain Jenkins-specific API classes would be moved into this plugin, so that it would not be a pure library wrapper.

yes see https://github.com/jenkinsci/kubernetes-plugin/pull/591#issuecomment-528884817  but that is not a good idea at all.  any util packages should go into a `kubernetes-client-utils-plugin` IMO

**Comment by cronik** [comment]
You guys have more at stake here so I'll defer to you on wrapper plugin conventions. 

I'm new to the jenkins plugin world so my interpretation of the [wrapper plugins](https://jenkins.io/doc/developer/plugin-development/dependencies-and-class-loading/#using-library-wrapper-plugins) docs was that this was a valid approach.

> Occasionally it can make sense to include a few API classes in the wrapper plugin itself, where any Jenkins code using the library would reasonably need some boilerplate adapters to standard Jenkins facilities.

I also consulted other plugins like [jenkinsci/git-client-plugin](https://github.com/jenkinsci/git-client-plugin) which reinforced my interpretation.

Based on this changeset, the wrapper plugin version convention is `${base library version}-${plugin increment}`?

**Comment by jglick** [comment]
Something like that. I think `apache-httpcomponents-client-4-api` is the most widely used example.

**Comment by Vlatombe** [comment]
Superceded by #4

**Pull request closed without merging**